dev/tests/unit/Magento/FunctionalTestFramework/Suite/SuiteGeneratorTest.php

```diff
@@ -143,6 +145,49 @@ public function testGenerateEmptySuite()
         $mockSuiteGenerator->generateSuite("basicTestSuite");
     }
 
+    public function testInvalidTestRef()
+    {
+        // Mock Suite1 => Test1 and Suite2 => Test2
+        $suiteDataArrayBuilder = new SuiteDataArrayBuilder();
+        $mockData = $suiteDataArrayBuilder
+            ->withName('Suite1')
+            ->includeGroups(['group1'])
+            ->build();
+        $suiteDataArrayBuilder = new SuiteDataArrayBuilder();
+        $mockData2 = $suiteDataArrayBuilder
+            ->withName('Suite2')
+            ->includeGroups(['group2'])
+            ->build();
+        $mockSuiteData = array_merge_recursive($mockData, $mockData2);
+
+        $testDataArrayBuilder = new TestDataArrayBuilder();
+        $mockSimpleTest = $testDataArrayBuilder
+            ->withName('Test1')
+            ->withAnnotations(['group' => [['value' => 'group1']]])
+            ->withTestActions()
+            ->build();
+        $testDataArrayBuilder = new TestDataArrayBuilder();
+        $mockSimpleTest2 = $testDataArrayBuilder
+            ->withName('Test2')
+            ->withAnnotations(['group' => [['value' => 'group2']]])
+            ->withTestActions()
+            ->build();
+        $mockTestData = ['tests' => array_merge($mockSimpleTest, $mockSimpleTest2)];
+        $this->setMockTestAndSuiteParserOutput($mockTestData, $mockSuiteData);
+
+        // Make invalid manifest
+        $suiteConfig = ['Suite2' => ['Test1']];
+        $manifest = TestManifestFactory::makeManifest('default', $suiteConfig);
+
+        // Set up Expected Exception
+        $this->expectException(TestReferenceException::class);
+        $this->expectExceptionMessageRegExp('(Suite: "Suite2" Tests: "Test1")');
+
+        // parse and generate suite object with mocked data and manifest
+        $mockSuiteGenerator = SuiteGenerator::getInstance();
+        $mockSuiteGenerator->generateAllSuites($manifest);
+    }
+
```

Comment:
This test is good for the 1st exception case. To test the 2nd exception case, you can use a non-existing suite name and you don't even need to mock any suite/test structure.